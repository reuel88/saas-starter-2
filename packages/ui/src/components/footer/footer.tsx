import { FC } from "react";
import { Github, Linkedin } from "lucide-react";
import { CompanyLogo } from "../company-logo";
import { Container } from "../container";
import { Wrapper } from "../wrapper";

export type FooterProps = {};

export const Footer: FC<FooterProps> = () => {
  return (
    <Wrapper as="footer" aria-labelledby="footer-heading">
      <Container className="py-16">
        <h2 id="footer-heading" className="sr-only">
          Footer
        </h2>

        <div className="xl:grid xl:grid-cols-3 xl:gap-8">
          <div className="space-y-8">
            <a href="/">
              <CompanyLogo />
            </a>
            <p className="text-sm leading-6">
              Making the world a better place through constructing elegant
              hierarchies.
            </p>
            <div className="flex space-x-6">
              <a
                href="#"
                className="text-foreground/40 hover:text-foreground/70"
                target="_blank"
              >
                <span className="sr-only">Github</span>
                <Github className="h-6 w-6" aria-hidden="true" />
              </a>

              <a
                href="#"
                className="text-foreground/40 hover:text-foreground/70"
                target="_blank"
              >
                <span className="sr-only">Linkedin</span>
                <Linkedin className="h-6 w-6" aria-hidden="true" />
              </a>
            </div>
          </div>
          <div className="mt-16 grid gap-y-16 sm:grid-cols-2 sm:gap-x-8 lg:grid-cols-4 lg:gap-y-8 xl:col-span-2 xl:mt-0">
            <div>Menu</div>
            <div>Menu</div>
            <div>Menu</div>
            <div>Menu</div>
          </div>
        </div>

        <div className="prose prose-slate mt-16 max-w-none space-y-4 border-t border-border pt-8 text-xs text-muted-foreground dark:prose-invert sm:mt-20 lg:mt-24">
          <p>&copy;{new Date().getFullYear()} Reuel. All right reserved</p>
        </div>
      </Container>
    </Wrapper>
  );
};
