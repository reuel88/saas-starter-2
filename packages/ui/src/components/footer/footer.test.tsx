import { render, screen } from "@testing-library/react";
import { Footer } from "./footer";

describe("CompanyLogo", () => {
  it("should render correctly", () => {
    // Arrange

    // Act
    const { baseElement } = render(<Footer />);
    const descriptionElement = screen.getByText(
      /Making the world a better place through constructing elegant hierarchies/,
    );

    // Assert
    expect(descriptionElement).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });
});
