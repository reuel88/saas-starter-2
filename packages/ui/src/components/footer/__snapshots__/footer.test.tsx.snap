// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CompanyLogo > should render correctly 1`] = `
<body>
  <div>
    <footer
      aria-labelledby="footer-heading"
      class="bg-background text-foreground"
    >
      <div
        class="mx-auto max-w-7xl px-6 lg:px-8 py-16"
      >
        <h2
          class="sr-only"
          id="footer-heading"
        >
          Footer
        </h2>
        <div
          class="xl:grid xl:grid-cols-3 xl:gap-8"
        >
          <div
            class="space-y-8"
          >
            <a
              href="/"
            >
              <div
                class="inline-block"
              >
                <span
                  class="sr-only"
                >
                  Blaiz Money
                </span>
                <svg
                  class="lucide lucide-flame h-9 w-9 text-hue-500"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8.5 14.5A2.5 2.5 0 0 0 11 12c0-1.38-.5-2-1-3-1.072-2.143-.224-4.054 2-6 .5 2.5 2 4.9 4 6.5 2 1.6 3 3.5 3 5.5a7 7 0 1 1-14 0c0-1.153.433-2.294 1-3a2.5 2.5 0 0 0 2.5 2.5z"
                  />
                </svg>
              </div>
            </a>
            <p
              class="text-sm leading-6"
            >
              Making the world a better place through constructing elegant hierarchies.
            </p>
            <div
              class="flex space-x-6"
            >
              <a
                class="text-foreground/40 hover:text-foreground/70"
                href="#"
                target="_blank"
              >
                <span
                  class="sr-only"
                >
                  Github
                </span>
                <svg
                  aria-hidden="true"
                  class="lucide lucide-github h-6 w-6"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M15 22v-4a4.8 4.8 0 0 0-1-3.5c3 0 6-2 6-5.5.08-1.25-.27-2.48-1-3.5.28-1.15.28-2.35 0-3.5 0 0-1 0-3 1.5-2.64-.5-5.36-.5-8 0C6 2 5 2 5 2c-.3 1.15-.3 2.35 0 3.5A5.403 5.403 0 0 0 4 9c0 3.5 3 5.5 6 5.5-.39.49-.68 1.05-.85 1.65-.17.6-.22 1.23-.15 1.85v4"
                  />
                  <path
                    d="M9 18c-4.51 2-5-2-7-2"
                  />
                </svg>
              </a>
              <a
                class="text-foreground/40 hover:text-foreground/70"
                href="#"
                target="_blank"
              >
                <span
                  class="sr-only"
                >
                  Linkedin
                </span>
                <svg
                  aria-hidden="true"
                  class="lucide lucide-linkedin h-6 w-6"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z"
                  />
                  <rect
                    height="12"
                    width="4"
                    x="2"
                    y="9"
                  />
                  <circle
                    cx="4"
                    cy="4"
                    r="2"
                  />
                </svg>
              </a>
            </div>
          </div>
          <div
            class="mt-16 grid gap-y-16 sm:grid-cols-2 sm:gap-x-8 lg:grid-cols-4 lg:gap-y-8 xl:col-span-2 xl:mt-0"
          >
            <div>
              Menu
            </div>
            <div>
              Menu
            </div>
            <div>
              Menu
            </div>
            <div>
              Menu
            </div>
          </div>
        </div>
        <div
          class="prose prose-slate mt-16 max-w-none space-y-4 border-t border-border pt-8 text-xs text-muted-foreground dark:prose-invert sm:mt-20 lg:mt-24"
        >
          <p>
            Â©
            2024
             Reuel. All right reserved
          </p>
        </div>
      </div>
    </footer>
  </div>
</body>
`;
